src/index.js

```diff
@@ -115,23 +120,14 @@ function deconstruct (selector, styles, media) {
 
     if (type === 'number' || type === 'string') {
       decs.push(createDec(key, value))
-      continue
     } else if (Array.isArray(value)) {
-      forEach(value, val => {
-        decs.push(createDec(key, val))
-      })
-      continue
-    } else if (/^:/.test(key)) {
+      forEach(value, val => decs.push(createDec(key, val)))
+    } else if (key.charCodeAt(0) === 58) {
       forEach(deconstruct(selector + key, value, media), r => rules.push(r))
-      continue
-    } else if (/^@media/.test(key)) {
+    } else if (key.indexOf('@media') !== -1) {
```

Comment:
I thought so too, but it can't. The bug happened because your media rules come in like this:

```js
console.log(`WAS MEDIA: ${key}`);
//=> WAS MEDIA:
          @media only screen and (-webkit-min-device-pixel-ratio: 1.5),
          only screen and (min--moz-device-pixel-ratio: 1.5),
          only screen and (-o-min-device-pixel-ratio: 1.5/1),
          only screen and (min-resolution: 144dpi),
          only screen and (min-resolution: 1.5dppx)
```

There's leading whitespace / multi-line, so your regex of `/^@media/` would always fail for large MQs.